### ISSUE TITLE
Host admin can freeze Collective activity

### ISSUE THREAD

**Issue by alanna** [opencollective/opencollective#4694]
## User story 
As a host admin, sometimes we need to flag a Collective due to various issues and not allow certain activities while we sort it out. Right now we have to manage this manually and it's really tricky in a large host with multiple staff processing expenses etc.

Examples I've dealt with:

* Expense payment is pending signing of a contract before we can release funds (it was a large payment to a for-profit company, which for OCF means we need a contract for services in place)
* Collective has committed the funds to a specific purpose and we can't let them spend them on something else (Collective is asking us to sign a lease on their behalf, requiring them to set aside the money to cover the value of the lease term and not spend it elsewhere)
* Internal conflict among Collective admins where we don't want them to add or remove admins until we work it out
* We are alerted to possible fraud or misuse of funds and need to freeze incoming contributions and outgoing expenses
* We become aware that a worker has to be classified as an employee and can't continue to pay them as a contractor, and need to stop payment on outstanding invoice expenses

## Best solution for this problem

![Screen Shot 2021-09-17 at 1 56 06 PM](https://user-images.githubusercontent.com/5498878/133712070-dac95dd9-1bf6-491b-b1cc-729c28158763.jpg)

A freeze menu in the hosted collectives dashboard screen with these options:

* Expenses (disables pay button on all outstanding and new expenses, but still lets them submit expenses)
  * Invoices (select)
  * Receipts (select)
* Admins (stops any removing or adding of admins)
* Contributions (disables new contributions and pauses recurring ones)
* A text box to write a note about what's going on

## MVP
Add these functions to the root admin dashboard in an MVP way. The most important one in my mind is expenses, so MVP could just focus on that.

**Comment by Betree** [comment]
@alanna I love the freeze icon :smile: Until we have something implemented here, feel free to ping engineering for such cases; we can manually disable features selectively (expenses, contributions, etc) for an account in the database.

On the API side, there's not much needed to implement this: we can rely on `account.features` to disable contributions and expenses. We'll probably need a new permission like `FEATURE.EDIT_MEMBERS` to prevent editing admins.

We'll also need an email template to inform collectives and send them the message (it should ideally details which features are blocked).

**Comment by aomuse** [comment]
Here to plus one and say @alanna is right. Use case is now proven out by a dispute in one of our current projects.

**Comment by kaylarep** [comment]
in the last week alone, OCF has had multiple internal disputes arise from within a few of our hosted collectives.  to me, this is the most immediate solution in the first step towards conflict resolution.

 feels pretty vital to have this - before things get bigger.

**Comment by Betree** [comment]
Just a note to say that I've implemented a quick flag to disable all features for a collective at once: https://github.com/opencollective/opencollective-api/pull/7187. We can probably re-use that to resolve this issue

**Comment by Betree** [comment]
# User needs

:art: = needs required

- :art: Confirmation modal
  - Checkbox to not send a notification
  - Ability to add a custom message if a notification is sent. We need to record this to be accessible to the admins (BCC all host admins, store it in the database > Activities)
- Notification
  - Include the custom message
  - "Your account has been frozen. Please contact your fiscal host or [support@opencollective.com](mailto:support@opencollective.com) for more information." (added the support email here in case we are freezing an independent collective)
  - Would be sent from no-reply@opencollective.com, but with a link to the contact page of the host: https://opencollective.com/opensource/contact
- :art: Have a special view for all collective pages to surface that the collective is frozen (kind of like what we do for achieved collectives). We must be careful with the wording and not disclose too much info, there are many reasons why a collective may be frozen. Maybe something like "Contributions to this page cannot be accepted at this time".
- Must be able to unfreeze, no need for a custom message there
- We must make sure that children (projects/events) are frozen as well
- :art: Would be great to have some granularity: eg. block expenses, but still allow to receive contributions/issue updates

**Comment by Memo-Es** [comment]
Is this issue replacing #3330 ? @alanna wdyt?

**Comment by natehn** [comment]
@Memo-Es They are different, and both are needed. Freezing is the more immediate need imo, but it is a preliminary step when dealing with problematic collectives. Then the host may choose to un-host, or unfreeze it and return to an active state.

**Comment by piamancini** [comment]
- [ ] Design Input

**Comment by natehn** [comment]
@Memo-Es I know you mentioned potentially changing the language to "pause" collective but in my opinion "freeze" is actually the proper English term. For example, the phrase "freeze" bank account is commonly used, and at a previous fiscal sponsor "freeze" was also used. It is not actually all that much like DnD :)

**Comment by kaylarep** [comment]
what about a stop sign?  (is a red octogon universal enough for 'Stop'? that hand looks like a wave to me...

**Comment by alanna** [comment]
I actually think a pause icon would be good. But I 100% agree we should
name it "freeze"*.*Message ID:
***@***.***>

>

**Comment by Memo-Es** [comment]
## Design update

| Freeze Collective icon  |  Confirmation modal + send message  | Collective page  |
|---|---|---|
| <img width="422" alt="Screen Shot on 2022-03-16 at 23-25-55" src="https://user-images.githubusercontent.com/49539954/158743137-df5b1c9c-7862-4cd9-ae3a-82e5f8070a58.png">  | ![Modal](https://user-images.githubusercontent.com/49539954/158743135-44c6aa87-d6d2-4cce-ad05-e88fd8cbe1f4.png)  |  ![Collective Page](https://user-images.githubusercontent.com/49539954/158743130-e7c3b62a-6d77-4735-8afa-7b41799af1f8.png) |

Feedback and help to nail down the final copy are welcome @alanna @natehn

**Comment by alanna** [comment]
Does freezing also include stopping contributions?

**Comment by natehn** [comment]
@alanna we discussed how a nice-to-have feature would be having a checkbox for whether or not to allow contributions, as shown in the design. I believe the MVP does not allow contributions. Does that feel right to you?

@Betree @SudharakaP - I'm don't think we want admins to be able to submit Updates in the MVP. In a more nuanced version, this could be its own option as well, alongside contributions.

@Memo-Es here are my edits to the copy:
- change "wont" to "won't" or "will not" and "payout" to "pay out"
- might remove note about Updates per above?
- I would change "Let the admins know how they can solve this situation" to "Let the admins know if any action is  required" ...or something like that

**Comment by natehn** [comment]
For the copy for the banner - what about "temporarily restricted" rather than just restricted? 

Another, softer term is "limited" - so perhaps it could be "temporarily limited"

**Comment by alanna** [comment]
If there could be options when the host freezes, i.e. whether to pause contributions, expenses, updates, that would be ideal.

**Comment by kaylarep** [comment]
## My take:

+1 to 'temporarily limited' over restricted

### on the popup:

> **Are you sure want to freeze this collective?**
> Freezing this collective means temporarily limiting what a collective (and their related Projects) can and cannot do on the platform.
> 
> They will **not** be able to accept funds, pay out expenses, or post updates. ^
> 
> **Include a message to the Collective admins (Optional)** They will also be notified of this freeze via auto-email notification.
> [Write an additional note to admins to let them know if action is required]
> 
> [ ] Allow frozen collective to receive contributions
> [ ] Allow frozen collective to pay out expenses
> [ ] Allow frozen collective to post updates

 
 ^ can they still edit their settings/team/about section/move money between projects...is there any other functionality we're forgetting to include/that they should or shouldn't be restricted from?)

**Comment by Memo-Es** [comment]
I think the granularity of the capabilities of frozen Collectives should be tackled in a follow-up project. Also, I'm not a fan of that many checkboxes in the modal.
Anyway, let's check with @Betree when he's back

**Comment by alanna** [comment]
> (and their related Projects)

And Events right?

What happens to their recurring contributions? Are they cancelled or just paused? I hope not cancelled.

I think three checkboxes is fine. They represent the real diversity of situations where we need to apply this action. This will be a rarely used feature only seen by host admins. We don't care if it's ugly or complicated. We care about more fine-grained control depending on the situation. If there are technical reasons it's too much to have all three options, and it can't be in scope, fine, but let's not make that call for UX reasons.

**Comment by Memo-Es** [comment]
If its a rarely used feature (I agree it is), perhaps it shouldn't be in your face all the time; then, we can put it under a 'More' icon instead of a more distracting one; it gives us the flexibility to add more stuff in there in the future.
@alanna

**Comment by Betree** [comment]
> I think the granularity of the capabilities of frozen Collectives should be tackled in a follow-up project. Also, I'm not a fan of that many checkboxes in the modal. Anyway, let's check with @Betree when he's back

Indeed. In the MVP we're going to start with blocking everything. We can then iterate to add checkboxes for specific features.

@Memo-Es The design shared in https://github.com/opencollective/opencollective/issues/4694#issuecomment-1070341817 looks good. We can foresee that we'll probably need something like a menu to handle all the specific actions (like #3330) in the future.

**Comment by kaylarep** [comment]
> I think the granularity of the capabilities of frozen Collectives should be tackled in a follow-up project. Also, I'm not a fan of that many checkboxes in the modal. 

am also fine with tackling this later.  but I was not involved in the most recent need-cases so much, so not sure how vital this granularity is immediately. (or if having this level of granularity would have been helpful).  I also feel like we're not covering all the right granularity in the 3 checkboxes, so i'd rather take the time to flesh it out properly.

__
 adding in the comments from above:

## on the popup:
Are you sure want to freeze this collective?
Freezing this collective means temporarily limiting what a collective (and their connected Projects & Events) can and cannot do on the platform.

They will not be able to accept funds, pay out expenses, post updates, create new Events or Projects under this collective.  However, they will still continue to receive recurring donations that were started before this freeze.

Include a message to the Collective admins (Optional) They will also be notified of this freeze via auto-email notification.
[Write an additional note to admins to let them know if action is required]

**Comment by kaylarep** [comment]
we also need consider the 'unfreezing' side of this. (or whatever we might want to call it)
[ ] play icon instead of pause icon
[ ] module popup
[ ] auto email to collective

on the popup:
Are you sure want to unfreeze this collective?
Unfreezing this collective means reinstating their complete usability on the platform.

This collective (and all its related Projects & Events) will now have access to accept funds, pay out expenses, post updates, create new Events or Projects. 

Include a message to the Collective admins (Optional) They will also be notified of this unfreeze via auto-email notification.
[Write an additional note to admins to let them know if action is required]

**Comment by Betree** [comment]
@kaylarep Thanks for the feedback on the wording, I applied your suggestions for the first implementation of the modal!

I need some more input for the emails sent to the collective admins. I created these first drafts:

**Frozen collective email**

> Hi {name},
> 
> Your fiscal host, Open Source Collective org, has frozen your Collective. In other words, they've temporarily limiting what you (and your connected Projects & Events) can and cannot do on the platform.
> 
> Your collective, {name}, will not be able to accept funds, pay out expenses, post updates, create new Events or Projects, add new Team members under this collective. However, they will still continue to receive recurring donations that were started before this freeze.
> 
> Open Source Collective org included the following message:
> ...

**Unfrozen collective email**

> Hi {name},
> 
> Your fiscal host, Open Source Collective, has unfrozen your Collective. In other words, they've temporarily limiting what you (and your connected Projects & Events) can and cannot do on the platform.
> 
> Your collective {name} (and all its related Projects & Events) will now have access to accept funds, pay out expenses, post updates, create new Events or Projects.
> 
> Open Source Collective org included the following message:
> ...

**Comment by natehn** [comment]
small edits in bold/strikethrough @Betree:

**Frozen:**

> ...they've temporarily ~~limiting~~ **limited**...

> ...**or** add new Team members under this collective.

^ noting that this list will need to change based on the check boxes

> However, ~~they~~ **the Collective** will

**Unfrozen:**

> In other words, they ~~'ve temporarily limiting what you (and your connected Projects & Events) can and cannot do on the platform.~~ **have reactivated any features that were previously limited while frozen.**

> ...~~will~~ **may** again ~~have access to~~ accept funds, pay out expenses, post updates, **and** create new Events or Projects.

^ again, adjusting based on granularity

Hope this is helpful!

**Comment by kaylarep** [comment]
One more line to add to the bottom of **Frozen Collective**:
> If you have additional questions about this freeze, please contact your fiscal host.

**Comment by piamancini** [comment]
- [ ] finish the front end 
- [ ] limit the features a host can block
- [ ] decide scope of the block

**Comment by SudharakaP** [comment]
> Include a message to the Collective admins (Optional) They will also be notified of this unfreeze via auto-email notification. 

@kaylarep : Here we use "notified/notification" twice which to me seems repetitive somehow. I think we could avoid this by using something like, "An automatic email notification will be sent to them about the unfreeze". WDYT? 🤔
